types/mdx/preact-tests.tsx

```diff
@@ -0,0 +1,571 @@
+import { MDXComponents, MDXContent, MDXModule } from "mdx/types";
+import { Component, ComponentProps, h } from "preact";
+import * as runtime from "preact/jsx-runtime";
+import MyMarkdownComponent from "./MyComponent.markdown";
+import MyMDComponent from "./MyComponent.md";
+import MyMDownComponent from "./MyComponent.mdown";
+import MyMDXComponent from "./MyComponent.mdx";
+import MyMKDComponent from "./MyComponent.mkd";
+import MyMKDNComponent from "./MyComponent.mkdn";
+import MyMKDownComponent from "./MyComponent.mkdown";
+import MyRonComponent from "./MyComponent.ron";
+
+// Test setup — User code
+
+declare module "mdx/types" {
+    namespace JSX {
+        type Element = runtime.JSX.Element;
+        type ElementClass = runtime.JSX.ElementClass;
+        type ElementType = runtime.JSX.ElementType;
+        type IntrinsicElements = runtime.JSX.IntrinsicElements;
+    }
```

Comment:
I was a bit sad to find out this is necessary. I initially thought this would work, but it doesn’t:

```ts
declare module 'mdx/types' {
  import { JSX } from 'preact/jsx-runtime'
}
```